stock_mts_mto_rule/models/stock_rule.py

```diff
@@ -68,6 +71,9 @@ def _run_split_procurement(self, procurements):
                 procurement.product_qty,
                 procurement.product_uom,
                 procurement.values,
+                done_proc_qty=done_procurements_by_product.get(
+                    procurement.product_id.id, 0.0
+                ),
```

Comment:
```suggestion
                qty_unavailable=qty_unavailable_by_product.get(
                    procurement.product_id.id, 0.0
                ),
```

Reply:
with a use of defaultdict it would be just
```suggestion
                done_proc_qty=done_procurements_by_product[procurement.product_id.id],
```